scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
    },
    "outputs": [],
    "source": [
-    "# Load the data into the dataset\n",
-    "ds[\"sample\"] = tiffs_to_variable(sample_path)\n",
-    "ds[\"reference\"] = tiffs_to_variable(ob_path)"
+    "# Adds a coordinate for the spectra and TOF\n",
+    "ds.coords[\"spectrum\"] = sc.Variable([\"spectrum\"], values=np.arange(ds[\"sample\"].shape[1]))\n",
+    "stitched = sc.Dataset(coords={\"tof\": sc.Variable([\"tof\"], unit=sc.units.us,\n",
+    "                              values=np.arange(start=rebin_parameters[\"start\"], stop=rebin_parameters[\"stop\"],\n",
+    "                                               step=rebin_parameters[\"width\"], dtype=np.float64))})"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": false
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "sc.plot.plot(sc.sum(ds, Dim.Spectrum)) if do_plots else None"
+    "# Pre-calculate frame shifts. This will be replaced with automatic WFM stitching in the future\n",
+    "frame_shifts = [sum(frame_shift_increments[:i + 1]) for i in range(len(frame_shift_increments))]\n",
+    "\n",
+    "def stitch_data(variable_to_stitch):\n",
+    "    print(f\"Stitching: {variable_to_stitch}\")\n",
+    "    stitched[variable_to_stitch] = imaging.stitch(ds[variable_to_stitch],\n",
+    "                                                  frame_parameters=frame_parameters,\n",
+    "                                                  frame_shifts=frame_shifts,\n",
+    "                                                  rebin_parameters=rebin_parameters)\n",
+    "\n",
+    "for var_name in ds.keys():\n",
+    "    stitch_data(var_name)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "# Adds a coordinate for the spectra\n",
-    "ds.coords[Dim.Spectrum] = sc.Variable([Dim.Spectrum], values=np.arange(ds[\"sample\"].shape[1]))\n",
-    "ds"
+    "integrated = sc.Dataset()\n",
+    "for dim in stitched.coords.keys():\n",
+    "    integrated.coords[dim] = stitched.coords[dim]\n",
+    "\n",
+    "integrated = sc.sum(stitched, 'tof')\n",
+    "# Put TOF coords back for the export to TIFF stack later\n",
+    "integrated.coords[\"tof\"] = stitched.coords[\"tof\"]"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# Pull out reference from out dataset to avoid checking for it in loops\n",
+    "reference = stitched[\"reference\"]\n",
+    "integrated_reference = integrated[\"reference\"]\n",
+    "\n",
+    "# Causes segmentation fault, see scipp/scipp issue: #1176\n",
+    "# when this can be added back we can remove the \"if \"reference\" checks later\n",
+    "# del stitched[\"reference\"]\n",
+    "# del integrated[\"reference\"]\n"
+   ],
```

Comment:
Why is all this needed?

Reply:
Originally it was to avoid the loops later on, but I ran into scipp/core #1176 

I've left it in so we can do things like
`integrated / integrated_reference`
`stitched / stitched_reference`

without dividing the reference by itself. Once the above issue is resolved we can uncomment it

Reply:
I don't know why this is related to the issue? I can't see what is going on.

Reply:
I'm trying to remove `reference` from the dataset at this point as in effect we have this happen later on if we don't

```
reference = d["reference"]
del d["reference"] # Cant do this atm
d / reference  # Divides by self so redundant 
```